lib/makeFormatter.js

```diff
@@ -1,506 +1,599 @@
+          nxt(2);
+          unshift('MMMM D, YYYY');
+          continue;
+        }
 
-    case 'l':
-      if(/^llll/.test(str)){
-        nxt(4);
-        unshift('ddd, MMM D, YYYY LT');
-        continue;
-      }
+        if (/^LTS/.test(str)) {
+          nxt(3);
+          unshift('h:mm:ss A');
+          continue;
+        }
 
-      if(/^lll/.test(str)){
-        nxt(3);
-        unshift('MMM D, YYYY LT');
-        continue;
-      }
+        if (/^LT/.test(str)) {
+          nxt(2);
+          unshift('h:mm A');
+          continue;
+        }
 
-      if(/^ll/.test(str)){
-        nxt(2);
-        unshift('MMM D, YYYY');
+        nxt(1);
+        unshift('MM/DD/YYYY');
         continue;
-      }
 
-      if(/^lts/.test(str)){
-        nxt(3);
-        unshift('h:mm:ss A');
+      case 'l':
+        if (/^llll/.test(str)) {
+          nxt(4);
+          unshift('ddd, MMM D, YYYY LT');
+          continue;
+        }
+
+        if (/^lll/.test(str)) {
+          nxt(3);
+          unshift('MMM D, YYYY LT');
+          continue;
+        }
+
+        if (/^ll/.test(str)) {
+          nxt(2);
+          unshift('MMM D, YYYY');
+          continue;
+        }
+
+        if (/^lts/.test(str)) {
+          nxt(3);
+          unshift('h:mm:ss A');
+          continue;
+        }
+
+        if (/^lt/.test(str)) {
+          nxt(2);
+          unshift('h:mm A');
+          continue;
+        }
+
+        nxt(1);
+        unshift('M/D/YYYY');
         continue;
-      }
 
-      if(/^lt/.test(str)){
-        nxt(2);
-        unshift('h:mm A');
+      case '[':
+        var idx = str.indexOf(']');
+        if (idx === -1) {
+          add('"["');
+          nxt(1);
+          continue;
+        }
+
+        add(JSON.stringify(str.slice(1, idx)));
+        nxt(idx + 1);
         continue;
-      }
-
-      nxt(1);
-      unshift('M/D/YYYY');
-      continue;
-
-    case '[':
-      var idx = str.indexOf(']');
-      if(idx === -1){
-        add('"["');
-        nxt(1); continue;
-      }
-
-      add(JSON.stringify(str.slice(1, idx)));
-      nxt(idx+1);
-      continue;
     }
 
     add(JSON.stringify(str.charAt(0)));
     nxt(1);
   }
 
-
   var fnBody = '';
 
-  if(gets.date){
+  if (gets.date) {
     fnBody += 'var _date = d.get' + (utc ? 'UTC' : '') + 'Date();\n';
   }
 
-  if(gets.month){
+  if (gets.month) {
     fnBody += 'var _month = d.get' + (utc ? 'UTC' : '') + 'Month();\n';
   }
 
-  if(gets.dayOfYear){
+  if (gets.dayOfYear) {
     fnBody +=
-          'var _startOfYear = new Date(d.valueOf());\n' +
-          '_startOfYear.set' + (utc ? 'UTC' : '') + 'Month(0);\n' +
-          '_startOfYear.set' + (utc ? 'UTC' : '') + 'Date(1);\n' +
-          'var _doy = Math.round((d - _startOfYear) / 864e5) + 1;\n';
+      'var _startOfYear = new Date(d.valueOf());\n' +
+      '_startOfYear.set' +
+      (utc ? 'UTC' : '') +
+      'Month(0);\n' +
+      '_startOfYear.set' +
+      (utc ? 'UTC' : '') +
+      'Date(1);\n' +
```

Comment:
Since we're adopting ES2015 minimum with the var -> const switch, it might be nice to take some of these places where prettier has split stuff up onto multiple lines and switch to using string interpolation to make it easier to follow.

So for example here we could do:
```suggestion
      `_startOfYear.set${utc ? 'UTC' : ''}Month(0);\n` +
      `_startOfYear.set${utc ? 'UTC' : ''}Date(1);\n` +
```

That way we're keeping all of the generated statements grouped together by line